app/src/main/java/com/pvp/app/ui/screen/calendar/MonthlyCalendarViewModel.kt

```diff
@@ -0,0 +1,114 @@
+package com.pvp.app.ui.screen.calendar
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.pvp.app.api.TaskService
+import com.pvp.app.api.UserService
+import com.pvp.app.model.Task
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import java.time.LocalDate
+import java.time.YearMonth
+import javax.inject.Inject
+
+@HiltViewModel
+class MonthlyCalendarViewModel @Inject constructor(
+    private val taskService: TaskService,
+    private val userService: UserService
+) : ViewModel() {
+    private val _currentYearMonth = MutableStateFlow(YearMonth.now())
+    private val _uiState = MutableStateFlow(CalendarUiState.Init)
+    val uiState: StateFlow<CalendarUiState> = _uiState.asStateFlow()
+
+    init {
+        viewModelScope.launch {
+            val flowUser = userService.getCurrent()
+            val flowTasks = flowUser.flatMapLatest { user ->
+                user
+                    ?.let { taskService.get(user.email) }
+                    ?: flowOf(listOf())
+            }
+
+            combine(flowUser, flowTasks) { user, tasks ->
+                _uiState.update { currentState ->
+                    currentState.copy(
+                        dates = getDates(currentState.yearMonth, tasks),
+                        tasks = tasks
+                    )
+                }
+            }.launchIn(viewModelScope)
+
+        }
+    }
+
+    fun changeMonth(nextMonth: YearMonth) {
+        _uiState.update {currentState ->
```

Comment:
Missing single space. Use Control+Alt+L for the whole file, it will fix all spacing problems for you.